lab_1/main.py

```diff
@@ -40,7 +70,18 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
     :param top_n: a number of the most common words
     :return: a list of the most common words
     """
-    pass
+
+    if isinstance(freq_dict, dict) and isinstance(top_n, int):
+        most_common = []
+        chart = sorted(list(freq_dict.values()), reverse=True)
+        chart = chart[:top_n]
+        freq_dict_temp = dict(freq_dict)
+        for i in chart:
+            word = list(freq_dict_temp.keys())[list(freq_dict_temp.values()).index(i)]
```

Comment:
hard to read
maybe you can sort by keys, not values, at the beginning? And there'd be no need for a loop?

Reply:
It is impossible in my code since sorting the keys results in alphabetical order rather than the sorting of frequencies. I see that the code can be shorter but not by that way. Or I need to rethink the algorithm.